src/Microsoft.Sbom.Api/Executors/Spdx3SerializationStrategy.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        return true;
+    }
+
+    public void AddToExternalDocRefsSupportingConfig(ref IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public async Task<List<FileValidationResult>> WriteJsonObjectsToSbomAsync(
+        ISbomConfig sbomConfig,
+        string spdxManifestVersion,
+        IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator,
+        IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator,
+        IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator,
+        IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator)
+    {
+        fileArrayGenerator.SbomConfig = sbomConfig;
+        packageArrayGenerator.SbomConfig = sbomConfig;
+        relationshipsArrayGenerator.SbomConfig = sbomConfig;
+        externalDocumentReferenceGenerator.SbomConfig = sbomConfig;
+
+        fileArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        packageArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        relationshipsArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        externalDocumentReferenceGenerator.SpdxManifestVersion = spdxManifestVersion;
+
+        // Files section
+        var generateResult = await fileArrayGenerator.GenerateAsync();
+        WriteElementsToSbom(generateResult);
+
+        // Packages section
+        var packagesGenerateResult = await packageArrayGenerator.GenerateAsync();
+        generateResult.Errors.AddRange(packagesGenerateResult.Errors);
+        WriteElementsToSbom(packagesGenerateResult);
+
+        // External Document Reference section
+        var externalDocumentReferenceGenerateResult = await externalDocumentReferenceGenerator.GenerateAsync();
+        generateResult.Errors.AddRange(externalDocumentReferenceGenerateResult.Errors);
+        WriteElementsToSbom(externalDocumentReferenceGenerateResult);
+
+        // Relationships section
+        var relationshipGenerateResult = await relationshipsArrayGenerator.GenerateAsync();
+        generateResult.Errors.AddRange(relationshipGenerateResult.Errors);
+        WriteElementsToSbom(relationshipGenerateResult);
+
+        return generateResult.Errors;
+    }
+
+    private void WriteElementsToSbom(GenerateResult generateResult)
+    {
+        // Write the JSON objects to the SBOM
+        foreach (var serializer in generateResult.SerializerToJsonDocuments.Keys)
+        {
+            // Write context
+            serializer.StartJsonArray("@context");
+            var document = JsonDocument.Parse(Constants.Spdx3Context);
+            serializer.Write(document);
+            serializer.EndJsonArray();
+
+            // Deduplication of elements by checking SPDX ID
+            var elementsSpdxIdList = new HashSet<string>();
+
+            serializer.StartJsonArray("@graph");
```

Comment:
"@graph" exists in 3 different classes--should it be a const that gets used everywhere?